src/main/java/duke/parser/Parser.java

```diff
@@ -0,0 +1,204 @@
+package duke.parser;
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import duke.DukeException;
+import duke.commands.ColourCommand;
+import duke.commands.Command;
+import duke.commands.DeadlineCommand;
+import duke.commands.DeleteCommand;
+import duke.commands.EventCommand;
+import duke.commands.ExitCommand;
+import duke.commands.FindCommand;
+import duke.commands.HelpCommand;
+import duke.commands.InvalidCommand;
+import duke.commands.ListCommand;
+import duke.commands.MarkCommand;
+import duke.commands.TodoCommand;
+import duke.commands.UnmarkCommand;
+import duke.common.Messages;
+
+/**
+ * Parses user input.
+ */
+public class Parser {
```

Comment:
To make your code easy to follow and readable, may be you can consider adding JavaDocs for all the functions in this class.